pennylane/beta/tapes/tape.py

```diff
@@ -904,6 +909,15 @@ def execute_device(self, params, device):
             params (list[Any]): The quantum tape operation parameters. If not provided,
                 the current tape parameter values are used (via :meth:`~.get_parameters`).
         """
+        if self._state_measurement:
+            if not device.capabilities().get("returns_state"):
+                raise ValueError("The current device is not capable of returning the state")
+            if len(self.measurements) > 1:
+                raise ValueError(
+                    "The state cannot be returned in combination with other return " "types"
```

Comment:
```suggestion
                    "The state cannot be returned in combination with other return types"
```

Reply:
Is this the right place to put device return validation? 🤔 It feels to me like it would better fit in the device/QubitDevice

Reply:
Have fixed the suggestion! Will come back to the question.

Reply:
Ok - I've moved the validation to the device: fe977b9249569a43d770e1c6bed9f6efd3b381d1
Looks better like this!